hooks/hooks.psm1

```diff
@@ -0,0 +1,731 @@
+$ErrorActionPreference = 'Stop'
+
+try {
+    $charmHelpersPath = Join-Path (Split-Path $PSScriptRoot) "lib\Modules\CharmHelpers"
+    Import-Module -Force -DisableNameChecking $charmHelpersPath
+} catch {
+    juju-log.exe "ERROR while loading PowerShell charm helpers: $_"
+    exit 1
+}
+
+$GIT_URL = "https://github.com/msysgit/msysgit/releases/download/Git-1.9.5-preview20150319/Git-1.9.5-preview20150319.exe"
+$GIT_SHA1 = "A8658BAE0DE8C8D3E40AA97A236A4FCF81DE50DF"
+$PYTHON27_URL = "https://www.python.org/ftp/python/2.7.10/python-2.7.10.msi"
+$PYTHON27_SHA1 = "9E62F37407E6964EE0374B32869B7B4AB050D12A"
+$7Z_URL = "http://www.7-zip.org/a/7z938.exe"
+$7Z_SHA1 = "9AC9E5E6A19BF3B18CD7BCBE34A5141996BB3028"
+
+$NEUTRON_GIT = "https://github.com/openstack/neutron.git"
+$NOVA_GIT = "https://github.com/openstack/nova.git"
+
+$OPENSTACK_DIR = Join-Path $env:SystemDrive "OpenStack"
+$PYTHON_DIR = Join-Path $env:SystemDrive "Python27"
+$BUILD_DIR = Join-Path $OPENSTACK_DIR "build"
+$INSTANCES_DIR = Join-Path $OPENSTACK_DIR "Instances"
+$BIN_DIR = Join-Path $OPENSTACK_DIR "bin"
+$CONFIG_DIR = Join-Path $OPENSTACK_DIR "etc"
+
+$NOVA_SERVICE_NAME = "nova-compute"
+$NOVA_SERVICE_DESCRIPTION = "OpenStack nova Compute Service"
+$NOVA_SERVICE_EXECUTABLE = Join-Path $PYTHON_DIR "Scripts\nova-compute.exe"
+$NOVA_SERVICE_CONFIG = Join-Path $CONFIG_DIR "nova.conf"
+
+$NEUTRON_SERVICE_NAME = "neutron-hyperv-agent"
+$NEUTRON_SERVICE_DESCRIPTION = "OpenStack Neutron Hyper-V Agent Service"
+$NEUTRON_SERVICE_EXECUTABLE = Join-Path $PYTHON_DIR "Scripts\neutron-hyperv-agent.exe"
+$NEUTRON_SERVICE_CONFIG = Join-Path $CONFIG_DIR "neutron_hyperv_agent.conf"
+
+$PYTHON_PROCESS_NAME = "python"
+
+$DOWNLOAD_LOCATION = "http://dl.openstack.tld/"
+
+
+function Get-TemplatesDir {
+    return (Join-Path (Get-JujuCharmDir) "templates")
+}
+
+
+function Get-DomainName {
```

Comment:
This function is of no use. Please use FQDN. Otherwise there is a high chance of name resolution failing.